src/commands/ping/ping.ts

```diff
@@ -1,21 +1,40 @@
-import { ApplyOptions } from '@sapphire/decorators';
-import { Command, CommandOptions, container } from '@sapphire/framework';
-import type { Message } from 'discord.js';
+import { Command, container } from '@sapphire/framework';
+import { Message } from 'discord.js';
 
-@ApplyOptions<CommandOptions>({
-  aliases: ['pong'],
-  description: 'ping pong',
-  detailedDescription: `**Examples:**
-\`${container.botPrefix}ping\`
-\`${container.botPrefix}pong\``
-})
-export class PingCommand extends Command {
-  async messageRun(message: Message): Promise<Message> {
-    const { client } = container;
-    const msg = await message.channel.send('Ping?');
-    const content = `Pong from JavaScript! Bot Latency ${Math.round(client.ws.ping)}ms. API Latency ${
-      msg.createdTimestamp - message.createdTimestamp
-    }ms.`;
-    return msg.edit(content);
+import { CodeyCommand, SapphireMessageExecuteType, SapphireMessageResponse } from '../../codeyCommand';
+
+const executeCommand: SapphireMessageExecuteType = (
+  client,
+  messageFromUser,
+  initialMessageFromBot
+): SapphireMessageResponse => {
+  // Assert message types are Message<boolean>
+  // We have to do this because APIMessage does not have "createdTimestamp" property
+  const messageFromUserAsMessage = <Message<boolean>>messageFromUser;
+  const initialMessageFromBotAsMessage = <Message<boolean>>initialMessageFromBot;
+
+  const botLatency = client.ws.ping;
+  const apiLatency = initialMessageFromBotAsMessage.createdTimestamp - messageFromUserAsMessage.createdTimestamp;
+  const content = `Pong from JavaScript! Bot Latency ${botLatency}ms. API Latency ${apiLatency}ms.`;
+  return content;
+};
+
+export class PingCommand extends CodeyCommand {
+  messageWhenExecutingCommand = 'Ping?';
+  messageIfFailure = 'Failed to receive ping.';
```

Comment:
Good to make this an option, but I am hoping that the response should be a generic "codey backend error" that we can standardize in utils.